picard/script/functions.py

```diff
@@ -286,6 +286,38 @@ def func_replace(parser, text, old, new):
     return text.replace(old, new)
 
 
+@script_function(documentation=N_(
+    """`$replacemulti(name,search,replace,separator="; ")`
+
+Replaces occurrences of `search` with `replace` in the multi-value tag `name`.
+
+Example:
+
+    $replacemulti(genre,Idm,IDM)
+"""
+))
+def func_replacemulti(parser, name, search, replace, separator=MULTI_VALUED_JOINER):
+    if not name or not separator:
```

Comment:
Should this also check for a missing `search` parameter, or do you want to allow replacement of empty list elements?

Reply:
I agree - `search` should be a required parameter. Indeed, I think that `replace` should be a required parameter, but it should be allowed to be empty.

i.e. `$replacemulti(%genre%)` and `$replacemulti(%genre%,search)` should both give an error, but `$replacemulti(%genre%,search,)` should not.

Reply:
Thanks both, this was an oversight on my part.

Reply:
What would be expected behaviour for an empty replacement ie. `$replacemulti(%genre%,search,)`? Should the search element be removed or set to the empty string?

Reply:
Good question.  I can see potential uses either way, but if I had to pick one I think I would go with removing them.  Perhaps there's a need for a new function `$cleanmulti()` to remove empty elements, because I'm not sure that there's currently an easy way to do that.  Then I would change my suggestion to leave them as empty strings.

Reply:
I've pushed some updates to addresses these issues. Currently the code sets an element to the empty string if replace is not specified, and I've added a test to confirm this behaviour.

Reply:
To me that makes sense